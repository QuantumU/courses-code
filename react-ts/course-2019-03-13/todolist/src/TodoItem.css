.todo {
  margin: 5px;
  padding: 5px;
  background-color: lightblue;
}

.todo.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}