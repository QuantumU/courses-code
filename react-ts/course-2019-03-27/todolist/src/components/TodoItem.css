.todo-item {
  background-color: lightgreen;
  padding: 8px;
  margin: 8px;
}

.todo-item.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}
