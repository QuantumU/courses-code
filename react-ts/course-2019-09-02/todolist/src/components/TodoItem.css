.todo {
  background-color: lightcoral;
  margin: 5px;
  padding: 5px;
}

.todo.completed {
  background-color: lightgrey;
}

.todo.completed>.label {
  text-decoration: line-through;
}
