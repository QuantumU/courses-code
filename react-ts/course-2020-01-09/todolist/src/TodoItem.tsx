import React, { useContext } from "react";
import ThemeContext from "./ThemeContext";

type TodoItemProps = {
  title: string;
  completed: boolean;
  onDelete: () => void;
  onToggle: () => void;
};

const TodoItem: React.FC<TodoItemProps> = props => {
  const theme = useContext(ThemeContext);
  return (
    <li
      onClick={() => props.onToggle()}
      className={props.completed ? "todoitem completed" : "todoitem"}
    >
      {props.completed ? "DONE: " : "TODO: "}
      {props.title}
      <button
        onClick={event => {
          // stop "event bubbling" to the parent element
          event.stopPropagation();
          props.onDelete();
        }}
        style={
          theme === "fancy"
            ? { textDecoration: "underline", backgroundColor: "pink" }
            : {}
        }
      >
        X
      </button>
    </li>
  );
};

export default TodoItem;
