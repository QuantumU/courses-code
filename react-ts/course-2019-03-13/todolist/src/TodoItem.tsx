import React, { memo, Component } from 'react';
import TodoType from './TodoType';

import './TodoItem.css';

type TodoItemProps = {
  todo: TodoType;
  onCompleted: () => void;
};

class TodoItem extends Component<TodoItemProps, {}> {

  shouldComponentUpdate(newProps: TodoItemProps) {
    return this.props.todo !== newProps.todo;
  }

  render = () => {
    return (
      <div
        className={`todo${this.props.todo.completed ? ' completed' : ''}`}
        onClick={() => {
          this.props.onCompleted();
        }}
      >
        {this.props.todo.completed ? 'DONE: ' : 'TODO: '}{this.props.todo.title}
      </div>
    );
  };
}

export default TodoItem;
