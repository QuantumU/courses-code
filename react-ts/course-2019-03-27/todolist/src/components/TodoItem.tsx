import React, { PureComponent } from 'react';
import TodoType from '../TodoType';
import './TodoItem.css';

type TodoItemProps = {
  todo: TodoType;
  onToggle: (id: number) => void;
};

type TodoItemState = {};

class TodoItem extends PureComponent<TodoItemProps, TodoItemState> {
  render() {
    // "todo-item" or "todo-item completed"
    return (
      <div
        className={
          this.props.todo.completed ? 'todo-item completed' : 'todo-item'
        }
        id="abc"
        onClick={() => {
          this.props.onToggle(this.props.todo.id);
        }}
      >
        {this.props.todo.title}
      </div>
    );
  }
}

export default TodoItem;
