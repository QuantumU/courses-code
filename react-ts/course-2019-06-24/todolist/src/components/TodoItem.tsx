import React from "react";
import { TodoType } from "../TodoType";
import "./TodoItem.css";

type TodoItemProps = {
  todo: TodoType;
  toggleTodo: (id: number) => void;
  deleteTodo: (id: number) => void;
};

const TodoItem = (props: TodoItemProps) => {
  return (
    <li
      key={props.todo.id}
      className={props.todo.completed ? "todo completed" : "todo"}
      onClick={() => {
        props.toggleTodo(props.todo.id);
      }}
    >
      {props.todo.completed ? "DONE: " : "TODO: "}
      {props.todo.title}
      <button onClick={() => {
        props.deleteTodo(props.todo.id)
      }}>X</button>
    </li>
  );
};

export default TodoItem;
