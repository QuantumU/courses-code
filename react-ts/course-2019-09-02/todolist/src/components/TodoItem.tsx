import React, { Component } from "react";
import TodoType from "../TodoType";
import "./TodoItem.css";
import Button from "@material-ui/core/Button";

type TodoItemProps = {
  todo: TodoType;
  onCompleted: (id: number) => void;
  onDelete: (id: number) => void;
};

class TodoItem extends Component<TodoItemProps> {
  render() {
    return (
      <div
        className={this.props.todo.completed ? "todo completed" : "todo"}
        onClick={this.handleClick}
      >
        <span className="label">
          {this.props.todo.completed ? "DONE" : "TODO"}: {this.props.todo.title}
        </span>{" "}
        <Button onClick={this.handleDelete} variant="contained" color="primary">
          X
        </Button>
      </div>
    );
  }

  handleClick = () => {
    this.props.onCompleted(this.props.todo.id);
  };

  handleDelete = (clickEvent: React.MouseEvent<HTMLButtonElement>) => {
    clickEvent.stopPropagation();
    this.props.onDelete(this.props.todo.id);
  };
}

export default TodoItem;
