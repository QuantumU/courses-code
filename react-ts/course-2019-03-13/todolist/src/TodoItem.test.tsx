import React from 'react';
import TodoItem from './TodoItem';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  shallow(
    <TodoItem
      todo={{ id: 1, title: 'abc', completed: false }}
      onCompleted={() => {}}
    />
  );
});

it('renders with correct content', () => {
  const todoItem = shallow(
    <TodoItem
      todo={{ id: 1, title: 'abc', completed: false }}
      onCompleted={() => {}}
    />
  );
  expect(todoItem.text()).toBe('TODO: abc');
});

it('renders with "todo" css class', () => {
  const todoItem = shallow(
    <TodoItem
      todo={{ id: 1, title: 'abc', completed: false }}
      onCompleted={() => {}}
    />
  );
  expect(todoItem.hasClass('todo')).toBeTruthy();
});

it('renders with "completed" css class', () => {
  const todoItem = shallow(
    <TodoItem
      todo={{ id: 1, title: 'abc', completed: true }}
      onCompleted={() => {}}
    />
  );
  expect(todoItem.hasClass('completed')).toBeTruthy();
});

it('reacts to clicks', () => {
  const mockFunction = jest.fn();
  const todoItem = shallow(
    <TodoItem
      todo={{ id: 1, title: 'abc', completed: true }}
      onCompleted={mockFunction}
    />
  );
  todoItem.simulate('click');
  expect(mockFunction).toBeCalledTimes(1);
});

it('matches snapshot 1', () => {
  const tree = renderer
    .create(<TodoItem
      todo={{ id: 1, title: 'abc', completed: true }}
      onCompleted={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
