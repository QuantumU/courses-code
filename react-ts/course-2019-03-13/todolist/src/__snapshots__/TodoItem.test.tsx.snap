// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="todo completed"
  onClick={[Function]}
>
  DONE: 
  abc
</div>
`;
